### PR TITLE
archive.extracted: don't try to cache local sources

### PR DESCRIPTION
**Pull request by terminalmage**
This will keep us from trying to cache file when we already have it locally,
which will help significantly with larger archives.

### CODE REVIEW

**Comment by rallytime** [comment]
@terminalmage Looks like there are some tests that need to be looked at here.

**Comment by terminalmage** [comment]
Yeah I only checked the integration tests, forgot to look at unit tests. This is likely just due to the logic changes since we're not caching the file any more. I'll get these taken care of.

**Comment by terminalmage** [comment]
Tests are updated

**Comment by terminalmage** [comment]
I've got a fix for the lint failures, waiting for the rest of the tests to complete before I push the lint fixes though.

**Comment by rallytime** [comment]
@terminalmage Hrm. It looks like some of these related tests are still failing on Ubuntu 14.

File: salt/states/archive.py [hunk 1]
```diff
@@ -1304,7 +1315,7 @@ def extracted(name,
                 ret['changes']['directories_created'] = [name]
             ret['changes']['extracted_files'] = files
             ret['comment'] = '{0} extracted to {1}'.format(source_match, name)
-            if not keep:
+            if not source_is_local and not keep:
```
Comment (ntrepid8) [inline salt/states/archive.py]: Am I reading this right that the default behavior now is to remove the local cached copy of a remove archive? It looks like the docs say the default for `keep` is supposed to be `True` but here it looks like the default is `False`.

I noticed a large increase in network activity and time when I updated some minions recently and tracked down the source to `archive.extracted`. The source files are no longer being stored in the minion cache and are being downloaded to the minions each time highstate is run.

Could this be the cause of the change in behavior?

Comment (terminalmage) [reply salt/states/archive.py]: @ntrepid8 The default behavior prior to 2017.7.3 would have been to remove the cached source archive. This was changed to keep the cached source in https://github.com/saltstack/salt/pull/43518 (backported for the 2017.7.3 release in https://github.com/saltstack/salt/pull/43681).

Comment (ntrepid8) [reply salt/states/archive.py]: :+1:

**Pull request merged**